### ISSUE TITLE
iOS : App crashes when Adhoc build is installed for the first time, works perfect with development certificates

### ISSUE THREAD

**Issue by sreejith-m-sreedharan** [phonegap/phonegap-plugin-push#744]
Hi, 

I am getting crash when build is taken with adHoc certificates,; With development certificates it works fine.

It crashes on app start, and sometimes it crashes when app comes to foreground.
It occurs one time only. After restarting app, it works fine. 

I googled the issue and came to know that someone got the same problem due to register calling more than once. I fixed the issue to ensure it calls only once. But even after that I am getting crash while installing adhoc build and using the app for a sometime.

#### Cordova CLI version and cordova platform version

    cordova --version                             6.0.0
    cordova platform version android     3.9.2

#### Plugin version
  Plugin version 1.4.5, 1.5.2

**Comment by macdonst** [comment]
@sreejith-m-sreedharan what version of cordova-ios are you using?

**Comment by ejfrias** [comment]
I'm also experiencing this issue. It seems like its getting stuck after calling init()

```
cordova -v 
6.0.0

cordova iOS version
4.0.1

phonegap-plugin-push
1.6.1

iOS Device version
9.3
```

**Comment by ejfrias** [comment]
Issue was fixed after upgrading cordova iOS version to 4.1

**Comment by macdonst** [comment]
@ejfrias yeah, I will be forcing everyone to use cordova-ios 4.1.x in release 1.7.0 of this plugin. 

Marking as duplicate of #751
